src/multiplex/client.rs

```diff
@@ -34,14 +35,118 @@ pub trait TagStore<Request, Response> {
     fn finish_tag(self: Pin<&mut Self>, r: &Response) -> Self::Tag;
 }
 
+/// A store used to track pending requests.
```

Comment:
Do we still need to impose `Tag: Eq`?

Reply:
Not following what you're asking here? Its imposed by the `TagStore` trait so we don't have much choice here.

Reply:
oh yup good call we can move that bound down into the impl of the pending store. Though I worry that this will cause some weird compiler errors maybe.

Reply:
```
error[E0599]: the method `build` exists for struct `multiplex::client::Builder<T, E, Request>`, but its trait bounds were not satisfied
   --> src/multiplex/client.rs:408:34
    |
75  | / pub struct VecDequePendingStore<T, Request>
76  | | where
77  | |     T: TryStream + Sink<Request> + TagStore<Request, T::Ok>,
78  | | {
79  | |     pending: VecDeque<Pending<T::Tag, T::Ok>>,
80  | |     _pd: PhantomData<fn((T, Request))>,
81  | | }
    | |_- doesn't satisfy `_: PendingStore<T, Request>`
...
292 | / pub struct Builder<
293 | |     T,
294 | |     E,
295 | |     Request,
...   |
302 | |     _pd: PhantomData<fn(Request, E)>,
303 | | }
    | |_- method `build` not found for this
...
408 |           Self::builder(transport).build()
    |                                    ^^^^^ method cannot be called on `multiplex::client::Builder<T, E, Request>` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `VecDequePendingStore<T, Request>: PendingStore<T, Request>`
note: the following trait must be implemented
   --> src/multiplex/client.rs:44:1
    |
44  | / pub trait PendingStore<T, Request>
45  | | where
46  | |     T: TryStream + Sink<Request> + TagStore<Request, T::Ok>,
47  | | {
...   |
65  | |     ) -> Result<Option<Pending<T::Tag, T::Ok>>, Error<T, Request>>;
66  | | }
    | |_^

error[E0599]: the function or associated item `new` exists for struct `multiplex::client::Builder<_, _, _, Box<(dyn FnOnce(E) + std::marker::Send + 'static)>, VecDequePendingStore<T, Request>>`, but its trait bounds were not satisfied
   --> src/multiplex/client.rs:413:89
    |
75  | / pub struct VecDequePendingStore<T, Request>
76  | | where
77  | |     T: TryStream + Sink<Request> + TagStore<Request, T::Ok>,
78  | | {
79  | |     pending: VecDeque<Pending<T::Tag, T::Ok>>,
80  | |     _pd: PhantomData<fn((T, Request))>,
81  | | }
    | |_- doesn't satisfy `_: PendingStore<_, _>`
...
292 | / pub struct Builder<
293 | |     T,
294 | |     E,
295 | |     Request,
...   |
302 | |     _pd: PhantomData<fn(Request, E)>,
303 | | }
    | |_- function or associated item `new` not found for this
...
413 |           Builder::<_, _, _, DefaultOnServiceError<E>, VecDequePendingStore<T, Request>>::new(
    |                                                                                           ^^^ function or associated item cannot be called on `multiplex::client::Builder<_, _, _, Box<(dyn FnOnce(E) + std::marker::Send + 'static)>, VecDequePendingStore<T, Request>>` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `VecDequePendingStore<T, Request>: PendingStore<_, _>`
note: the following trait must be implemented
   --> src/multiplex/client.rs:44:1
    |
44  | / pub trait PendingStore<T, Request>
45  | | where
46  | |     T: TryStream + Sink<Request> + TagStore<Request, T::Ok>,
47  | | {
...   |
65  | |     ) -> Result<Option<Pending<T::Tag, T::Ok>>, Error<T, Request>>;
66  | | }
    | |_^
```

Reply:
https://github.com/tower-rs/tokio-tower/pull/29/commits/181fdfb6785f8404bd5fe1d063592b3d946f9469 fixes this